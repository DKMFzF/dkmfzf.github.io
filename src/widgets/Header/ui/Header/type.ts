export type TIconDataLinks = Partial<{
	email: string;
	github: string;
	telegram: string;
}>;
