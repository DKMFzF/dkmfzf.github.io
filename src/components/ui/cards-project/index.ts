export { CardsProjectUI } from './cards-project';
