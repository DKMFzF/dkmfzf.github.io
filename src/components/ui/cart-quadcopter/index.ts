export { CartQuadcopter } from './cart-quadcopter';
