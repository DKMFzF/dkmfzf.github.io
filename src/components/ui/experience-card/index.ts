export { ExperienceCardUI } from './eperience-card';
