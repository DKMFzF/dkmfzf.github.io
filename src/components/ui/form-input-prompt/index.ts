export { FormInput } from './from-input';
