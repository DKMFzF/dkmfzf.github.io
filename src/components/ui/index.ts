export * from './preloader';
export * from './modal-overlay';
export * from './pages';
export * from './app-header';
export * from './modal';
export * from './icon';
export * from './link';
export * from './section-about-me';
