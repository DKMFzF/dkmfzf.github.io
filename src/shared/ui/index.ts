export * from './CommonPage';
export * from './Icon';
export * from './Preloader';
// export * from './Text';
