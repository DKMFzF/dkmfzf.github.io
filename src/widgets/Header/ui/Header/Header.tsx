import { FC } from 'react';
import { AnimatedNavLink } from '@ui';
import { useMobileMenu } from '../../model';
import { useMetrika } from '@lib';
import resumeFile from '../../../../../public/kirill-doroshev-resume.pdf';
import styles from './Header.module.scss';
import { useLocation } from 'react-router-dom';

// TODO: Распил компонента
// TODO: Выделить данные с хуков в пропсы и кинуть на уровень выше

export const Header: FC = () => {
	const {
		isMobileMenuOpen,
		menuRef,
		mobileMenuContentRef,
		overlayRef,
		toggleMenu,
		closeMenu
	} = useMobileMenu();

	const location = useLocation();

	const shouldApplyEmptyStyle =
		location.pathname === '/portfolio' ||
		location.pathname.startsWith('/sketches/');

	const { ym, gtag } = useMetrika();

	return (
		<header className={styles.header}>
			<div
				className={`${styles['header__container-default']}
				${shouldApplyEmptyStyle ? '' : styles.header__container}
			`}
			>
				<div className={styles['header__inside-wrapper']}>
					<div className={styles.header__brand}>
						<AnimatedNavLink
							to='/'
							styles={styles.header__link}
							data-text='portfolio'
							data-cy='link-portfolio'
						>
							portfolio
						</AnimatedNavLink>
					</div>

					<nav className={styles.header__nav}>
						<ul className={styles.header__menu}>
							<li className={styles.header__menuItem}>
								<AnimatedNavLink
									to='/'
									styles={styles.header__link}
								>
									about Me
								</AnimatedNavLink>
							</li>
							<li className={styles.header__menuItem}>
								<AnimatedNavLink
									to='/sketches'
									styles={styles.header__link}
								>
									sketches
								</AnimatedNavLink>
							</li>
							<li className={styles['header__menu-item']}>
								<AnimatedNavLink
									to='/portfolio'
									styles={styles.header__link}
									data-cy='link-portfolio'
								>
									portfolio
								</AnimatedNavLink>
							</li>
							<li className={styles['header__menu-item']}>
								<a
									href={resumeFile}
									download='kirill-doroshev-resume.pdf'
									className={styles.header__link}
									// SEO Analysis
									onClick={() => {
										ym('reachGoal', 'download-resume');
										gtag('event', 'file_download', {
											event_name: 'file_download'
										});
									}}
								>
									download resume
								</a>
							</li>
						</ul>
					</nav>

					<div
						className={styles['header__mobile-menu']}
						ref={menuRef}
					>
						<button
							className={styles['header__burger-button']}
							onClick={toggleMenu}
						>
							☰
						</button>

						{isMobileMenuOpen && (
							<>
								<div
									className={styles.header__overlay}
									ref={overlayRef}
									onClick={closeMenu}
								/>
								<nav
									className={
										styles['header__mobile-menu-content']
									}
									ref={mobileMenuContentRef}
								>
									<AnimatedNavLink
										to='/'
										styles={styles.header__link}
										onClick={closeMenu}
									>
										About Me
									</AnimatedNavLink>

									<AnimatedNavLink
										to='/sketches'
										styles={styles.header__link}
										onClick={closeMenu}
									>
										sketches
									</AnimatedNavLink>

									<AnimatedNavLink
										to='/portfolio'
										styles={styles.header__link}
										onClick={() => closeMenu()}
										data-cy='link-portfolio'
									>
										portfolio
									</AnimatedNavLink>

									<a
										href={resumeFile}
										download='kirill-doroshev-resume.pdf'
										className={styles.header__link}
										onClick={() => {
											ym('reachGoal', 'download-resume');
											gtag('event', 'file_download', {
												event_name: 'file_download'
											});
											closeMenu();
										}}
									>
										download resume
									</a>
								</nav>
							</>
						)}
					</div>
				</div>
			</div>
		</header>
	);
};
