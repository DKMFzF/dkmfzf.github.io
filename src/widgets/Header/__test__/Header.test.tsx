import { Link, MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import { Header } from '../ui/Header';
import { SocialLinks } from '@providers';
import { expect } from '@jest/globals';

jest.mock(
	'../../../../public/kirill-doroshev-resume.pdf',
	() => 'test-resume.pdf'
);

jest.mock('@ui', () => ({
	__esModule: true,
	default: () => <div data-testid='icon' />,
	Icon: () => <div data-testid='icon' />,
	AnimatedNavLink: ({ children }: any) => (
		<div data-testid='animated-link'>{children}</div>
	)
}));

describe('[Header]', () => {
	it('renders header correctly and matches snapshot', () => {
		const mockLinks = {
			github: 'https://github.com/test',
			email: 'test@example.com',
			telegram: 'https://t.me/test'
		};

		const { container } = render(
			<SocialLinks.Provider value={mockLinks}>
				<MemoryRouter>
					<Header />
				</MemoryRouter>
			</SocialLinks.Provider>
		);

		expect(container).toMatchSnapshot();
	});
});
