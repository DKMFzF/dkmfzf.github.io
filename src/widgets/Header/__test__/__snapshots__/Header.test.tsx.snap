// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Header] renders header correctly and matches snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <div
      class="header__container-default
				header__container
			"
    >
      <div
        class="header__inside-wrapper"
      >
        <div
          class="header__brand"
        >
          <span
            class="header__link"
            style="cursor: pointer;"
          >
            portfolio
          </span>
        </div>
        <nav
          class="header__nav"
        >
          <ul
            class="header__menu"
          >
            <li
              class="header__menuItem"
            >
              <span
                class="header__link"
                style="cursor: pointer;"
              >
                about Me
              </span>
            </li>
            <li
              class="header__menuItem"
            >
              <span
                class="header__link"
                style="cursor: pointer;"
              >
                sketches
              </span>
            </li>
            <li
              class="header__menu-item"
            >
              <span
                class="header__link"
                style="cursor: pointer;"
              >
                portfolio
              </span>
            </li>
            <li
              class="header__menu-item"
            >
              <a
                class="header__link"
                download="kirill-doroshev-resume.pdf"
                href="test-resume.pdf"
              >
                download resume
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="header__mobile-menu"
        >
          <button
            class="header__burger-button"
          >
            â˜°
          </button>
        </div>
      </div>
    </div>
  </header>
</div>
`;
